x/mongo/driver/operation/hello.go

```diff
@@ -171,14 +173,21 @@ func getFaasEnvName() string {
 	names := make(map[string]struct{})
 
 	for _, envVar := range envVars {
-		if os.Getenv(envVar) == "" {
+		val := os.Getenv(envVar)
+		if val == "" {
 			continue
 		}
 
 		var name string
 
 		switch envVar {
-		case envVarAWSExecutionEnv, envVarAWSLambdaRuntimeAPI:
+		case envVarAWSExecutionEnv:
+			if !strings.HasPrefix(val, awsLambdaPrefix) {
```

Comment:
Will the casing of the prefix be consistent? Or should we lowercase both before comparison?

Reply:
The [AWS documentation](https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html#configuration-envvars-runtime)and the [DBX specifications](https://github.com/mongodb/specifications/pull/1414/files#diff-0ecfac0976ff0bfe5b91b5bd52bc5a79dd1125434505ff22331c7cadf1b2f25eR330) do not mention case sensitivity. The [PyMongo implementation](https://github.com/ShaneHarvey/mongo-python-driver/blob/55fd51520e49efe0298085d8ea83bad9ce677d77/pymongo/pool.py#L240) does not check for case sensitivity. I don't see a definitive either way in the AWS docs, but they do say 

> AWS_EXECUTION_ENV – The [runtime identifier](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html), prefixed by AWS_Lambda_ (for example, AWS_Lambda_java8).

Reply:
Seems reasonable to keep it consistent with other implementations unless we discover an issue. Thanks for the detailed feedback!